common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -85,224 +105,44 @@ def default(self, obj):  # pylint: disable=method-hidden
-        values={"min": 0, "step": .1},
-        scope=Scope.settings
-    )
-    markdown = String(help=_("Markdown source of this module"), default=None, scope=Scope.settings)
-    source_code = String(
-        help=_("Source code for LaTeX and Word problems. This feature is not well-supported."),
-        scope=Scope.settings
-    )
-    use_latex_compiler = Boolean(
-        help=_("Enable LaTeX templates?"),
-        default=False,
-        scope=Scope.settings
-    )
-    matlab_api_key = String(
-        display_name=_("Matlab API key"),
-        help=_("Enter the API key provided by MathWorks for accessing the MATLAB Hosted Service. "
-               "This key is granted for exclusive use by this course for the specified duration. "
-               "Please do not share the API key with other courses and notify MathWorks immediately "
-               "if you believe the key is exposed or compromised. To obtain a key for your course, "
-               "or to report an issue, please contact moocsupport@mathworks.com"),
-        scope=Scope.settings
-    )
-
-
-class CapaMixin(ScorableXBlockMixin, CapaFields):
+class CapaMixin(ScorableXBlockMixin):
     """
         Core logic for Capa Problem, which can be used by XModules or XBlocks.
     """
-    def __init__(self, *args, **kwargs):
-        super(CapaMixin, self).__init__(*args, **kwargs)
-
+    @property
+    def close_date(self):
+        """
+        Return the date submissions should be closed from.
+        """
         due_date = self.due
 
         if self.graceperiod is not None and due_date:
-            self.close_date = due_date + self.graceperiod
+            return due_date + self.graceperiod
         else:
-            self.close_date = due_date
+            return due_date
 
+    def get_seed(self):
+        """
+        Generate the seed if not set and return it.
+        """
         if self.seed is None:
             self.choose_new_seed()
+        return self.seed
 
-        # Need the problem location in openendedresponse to send out.  Adding
-        # it to the system here seems like the least clunky way to get it
-        # there.
-        self.runtime.set('location', text_type(self.location))
-
+    @cached_property
+    def lcp(self):
         try:
-            # TODO (vshnayder): move as much as possible of this work and error
-            # checking to descriptor load time
-            self.lcp = self.new_lcp(self.get_state_for_lcp())
-
-            # At this point, we need to persist the randomization seed
-            # so that when the problem is re-loaded (to check/view/save)
-            # it stays the same.
-            # However, we do not want to write to the database
-            # every time the module is loaded.
-            # So we set the seed ONLY when there is not one set already
-            if self.seed is None:
-                self.seed = self.lcp.seed
```

Comment:
At the time this was added in https://github.com/edx/edx-platform/commit/fb9760244262073384c90ec0de29e737455c5ce8, seed was not generated for all cases in `CapaModule`: https://github.com/edx/edx-platform/blob/fb9760244262073384c90ec0de29e737455c5ce8/common/lib/xmodule/xmodule/capa_module.py#L137-L142

And could be generated in: https://github.com/edx/edx-platform/blob/fb9760244262073384c90ec0de29e737455c5ce8/common/lib/capa/capa/capa_problem.py#L119-L121

But all cases were moved to `CapaModule` in https://github.com/edx/edx-platform/commit/477fe670dd1b30ae97de3258cc8710255941c4a4#diff-54270f54a6e2a6fcbb04e74d13e191a3. 

So we not longer need to do this.